auth0/data_source_auth0_tenant.go

```diff
@@ -0,0 +1,40 @@
+package auth0
+
+import (
+	"fmt"
+	"net/url"
+
+	"github.com/auth0/go-auth0/management"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func newDataTenant() *schema.Resource {
+	return &schema.Resource{
+		Read: readDataTenant,
+		Schema: map[string]*schema.Schema{
+			"domain": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"management_api_identifier": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func readDataTenant(d *schema.ResourceData, m interface{}) error {
+	api := m.(*management.Management)
+
+	u, err := url.Parse(api.URI())
+	if err != nil {
+		return fmt.Errorf("unable to determine management API URL: %w", err)
+	}
+
+	d.SetId(resource.UniqueId())
+	d.Set("domain", u.Hostname())
+	d.Set("management_api_identifier", u.String())
```

Comment:
Let's make sure we don't ignore these errors. We can make use of the multierror pkg in this case. e.g.

```
result := multierror.Append(d.Set("domain", u.Hostname() ....)

return result.ErrorOrNil()